pkgs/by-name/bk/BK7231GUIFlashTool/package.nix

```diff
@@ -0,0 +1,94 @@
+{
+  lib,
+  stdenv,
+  fetchFromGitHub,
+  mono,
+  gtk2,
+  imagemagick,
+  makeWrapper,
+  copyDesktopItems,
+  makeDesktopItem,
+}:
+
+stdenv.mkDerivation rec {
+  pname = "BK7231GUIFlashTool";
+  version = "50";
+
+  src = fetchFromGitHub {
+    owner = "openshwprojects";
+    repo = "BK7231GUIFlashTool";
+    tag = "v${version}";
+    hash = "sha256-ssL5hX64maAl9qrJ08nUGfEWWLGfWbyv8St5WD4If2Y=";
+  };
+
+  nativeBuildInputs = [
+    mono
+    imagemagick
+    makeWrapper
+    copyDesktopItems
+  ];
+
+  buildInputs = [
+    gtk2
+  ];
+
+  buildPhase = ''
+    runHook preBuild
+    xbuild BK7231Flasher.sln /p:Configuration=Release
+    runHook postBuild
+  '';
+
+  installPhase = ''
+    runHook preInstall
+    mkdir -p $out/bin
+    mkdir -p $out/lib/BK7231GUIFlashTool
+    cp -r BK7231Flasher/bin/Release/* $out/lib/BK7231GUIFlashTool/
+    makeWrapper ${mono}/bin/mono $out/bin/BK7231Flasher \
+      --add-flags "$out/lib/BK7231GUIFlashTool/BK7231Flasher.exe" \
+      --prefix LD_LIBRARY_PATH : "${lib.makeLibraryPath [ gtk2 ]}"
+    runHook postInstall
+  '';
+
+  meta = with lib; {
```

Comment:
nit: I think a top level `with lib;` is generally discouraged [source](https://github.com/NixOS/nixpkgs/pull/293767)